src/uri/include/sourcemeta/jsontoolkit/uri.h

```diff
@@ -288,8 +288,25 @@ class SOURCEMETA_JSONTOOLKIT_URI_EXPORT URI {
   /// const sourcemeta::jsontoolkit::URI uri{
   ///   sourcemeta::jsontoolkit::URI::from_fragment("foo")};
   /// assert(uri.recompose() == "#foo");
+  /// ```
   static auto from_fragment(std::string_view fragment) -> URI;
 
+  /// Get the user information part of the URI, if any. For example:
+  ///
+  /// ```cpp
+  /// #include <sourcemeta/jsontoolkit/uri.h>
+  /// #include <cassert>
+  ///
+  /// const sourcemeta::jsontoolkit::URI uri{"https://user:@host"};
+  /// assert(uri.user_info().has_value());
+  /// assert(uri.user_info().value() == "user:);
+  /// ```
+  ///
+  /// As mentioned in RFC 3986, the format "user:password" is deprecated.
+  /// Applications should not render as clear text any data after the first
+  /// colon. See https://tools.ietf.org/html/rfc3986#section-3.2.1
+  [[nodiscard]] auto user_info() const -> std::optional<std::string_view>;
```

Comment:
Can we call this `userinfo`? The RFC refers to it as such, even though it uses hyphens for other parts of the grammar. Maybe worth sticking with that convention?

Reply:
Hm, I don't have strong opinions actually. This looks good too. I don't know

Reply:
done